rct_optimizations/CMakeLists.txt

```diff
@@ -42,48 +22,32 @@ add_library(${PROJECT_NAME}
   src/${PROJECT_NAME}/pnp.cpp
   src/${PROJECT_NAME}/multi_camera_pnp.cpp
 )
-
-add_dependencies(${PROJECT_NAME} ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
-
-target_link_libraries(${PROJECT_NAME}
-  ${catkin_LIBRARIES}
+target_compile_options(${PROJECT_NAME} PUBLIC -std=c++11)
+target_compile_options(${PROJECT_NAME} PRIVATE -Wall -Wextra)
+target_include_directories(${PROJECT_NAME} PUBLIC
+  "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+  "$<INSTALL_INTERFACE:include>")
+target_include_directories(${PROJECT_NAME} SYSTEM PUBLIC
+  ${CERES_INCLUDE_DIRS}
+)
+target_link_libraries(${PROJECT_NAME} PUBLIC
   ${CERES_LIBRARIES}
 )
 
-#############
-## Testing ##
-#############
-if(CATKIN_ENABLE_TESTING)
-  # Build the test support library
-  add_library(${PROJECT_NAME}_test_support
-    test/src/observation_creator.cpp
-    test/src/utilities.cpp)
-  target_link_libraries(${PROJECT_NAME}_test_support ${PROJECT_NAME})
-  target_include_directories(${PROJECT_NAME}_test_support PUBLIC test/include)
-
-  # The actual tests...
-  catkin_add_gtest(${PROJECT_NAME}_conversion_tests test/conversion_utest.cpp)
-  target_link_libraries(${PROJECT_NAME}_conversion_tests ${PROJECT_NAME}_test_support)
-
-  catkin_add_gtest(${PROJECT_NAME}_extrinsic_multi_static_camera_tests test/extrinsic_multi_static_camera_utest.cpp)
-  target_link_libraries(${PROJECT_NAME}_extrinsic_multi_static_camera_tests ${PROJECT_NAME}_test_support)
-
-  catkin_add_gtest(${PROJECT_NAME}_extrinsic_camera_on_wrist_tests test/extrinsic_camera_on_wrist_utest.cpp)
-  target_link_libraries(${PROJECT_NAME}_extrinsic_camera_on_wrist_tests ${PROJECT_NAME}_test_support)
+if(RCT_ENABLE_RUN_TESTING)
+  enable_testing()
+  rct_add_run_tests_target()
+  add_subdirectory(test)
 endif()
```

Comment:
I would also consider separating the running of the unit tests from the building of the tests. I'm finding right now that I want to build the tests often and run individual tests offline. This can be problematic because but they take 2-3 minutes in total to run. I don't want to wait that long every time I build unit tests just to see ones that I don't care about at the time run

Reply:
The macros do support separately building and running tests, I just need to change where they're included in relation to the `RCT_ENABLE_RUN_TESTING` flag.

One challenge is that I have to make sure that GTest can be found and linked against whenever we build the tests, which is a little tricky because its authors have conceptually designed it to need to be downloaded and built from source, vs. providing precompiled binaries.

Reply:
This should be addressed in my latest changes.